### DISCUSSION TITLE
Expected scalar type Double but found Float

### DISCUSSION THREAD

**Discussion by mariummou** [mir-group/allegro#101] [Q&A]
Hello,
I am trying use allegro right now and my config file has stressforce output model. I tried to do both NPT and NVT test using the pth deployed model. but getting the same error. Please let me know how to fix this. 
Exception: expected scalar type Double but found Float
Exception raised from check_type at /opt/conda/conda-bld/pytorch_1720538439675/work/build/aten/src/ATen/core/TensorMethods.cpp:12 (most recent call first):
frame #0: c10::Error::Error(c10::SourceLocation, std::string) + 0x96 (0x149976d91f86 in /usr/local/usrapps/dwb/mmou/env_nequip/lib/python3.12/site-packages/torch/lib/libc10.so)
frame #1: c10::detail::torchCheckFail(char const*, char const*, unsigned int, std::string const&) + 0x64 (0x149976d40d10 in /usr/local/usrapps/dwb/mmou/env_nequip/lib/python3.12/site-packages/torch/lib/libc10.so)
frame #2: <unknown function> + 0x322c752 (0x1499bcfa0752 in /usr/local/usrapps/dwb/mmou/env_nequip/lib/python3.12/site-packages/torch/lib/libtorch_cpu.so)
frame #3: double* at::TensorBase::mutable_data_ptr<double>() const + 0x43 (0x1499bcfa1803 in /usr/local/usrapps/dwb/mmou/env_nequip/lib/python3.12/site-packages/torch/lib/libtorch_cpu.so)
frame #4: at::TensorAccessor<double, 2ul, at::DefaultPtrTraits, long> at::TensorBase::accessor<double, 2ul>() const & + 0x34 (0x6339c4 in ./lmp_alg)
frame #5: ./lmp_alg() [0x8acb32]
frame #6: ./lmp_alg() [0x5b36d1]
frame #7: ./lmp_alg() [0x53e73a]
frame #8: ./lmp_alg() [0x4194f4]
frame #9: ./lmp_alg() [0x4181be]
frame #10: ./lmp_alg() [0x4149ec]
frame #11: <unknown function> + 0x3feb0 (0x14997663feb0 in /lib64/libc.so.6)
frame #12: __libc_start_main + 0x80 (0x14997663ff60 in /lib64/libc.so.6)
frame #13: ./lmp_alg() [0x4148c5]

Abort(1) on node 0 (rank 0 in comm 0): application called MPI_Abort(MPI_COMM_WORLD, 1) - process 0

**Comment by Linux-cpp-lisp** [comment]
Hi @mariummou ,

Please ensure you are using the latest `nequip` and latest `pair_allegro`, both on `main`.

**Reply by DQM520** [reply]
I met the same problem，not only using the colab tutorial but also in my own case:**MPI_ABORT was invoked on rank 0 in communicator MPI_COMM_WORLD
with errorcode 1.

NOTE: invoking MPI_ABORT causes Open MPI to kill all MPI processes.
You may or may not see output from other processes, depending on
exactly when Open MPI kills them.** Does it mean I need to use the latest nequip and pair_allegro?
![277fe1268f4f66ddbd06422f4e9b47e](https://github.com/user-attachments/assets/ef98e43d-e8c3-4fd1-ba8a-6b0338cd1807)
![73e21d165558448029c2490d82f7a31](https://github.com/user-attachments/assets/e5e0b696-8922-4c63-945b-d63eaa2f0349)
